### PR TITLE
show how to use a variable as hash map key

### PR DESCRIPTION
**Pull request by tolry**
currently the documentation does not explain how it's possible to use a variable as key in a hash map

### CODE REVIEW

File: doc/templates.rst [hunk 1]
```diff
@@ -604,6 +604,10 @@ exist:
 
     {# keys as expressions (the expression must be enclosed into parentheses) #}
     { (1 + 1): 'foo', (a ~ 'b'): 'bar' }
+    
+    {# keys as variables #}
+    {% set dynamic_key = 'foo' %}
+    { (dynamic_key): 'foo' }
```
Comment (fabpot) [inline doc/templates.rst]: That's the exact same example as the one above. What about merging the two with 3 examples: a "simple" var, an operation (1 + 1) and a concatenation (a ~ 'b')?

Comment (tolry) [inline doc/templates.rst]: sounds good, will get onto that

**Comment by tolry** [comment]
@fabpot updated, any thoughts? The second comment line might be a bit redundant, since expressions have their own chapter.

**Comment by tolry** [comment]
found a more compact solution

**Comment by fabpot** [comment]
Thank you @tolry.

**Pull request closed without merging**